storage/innobase/xtrabackup/src/xbstream_read.c

```diff
@@ -178,39 +188,30 @@ xb_stream_read_chunk(xb_rstream_t *stream, xb_rstream_chunk_t *chunk)
 	stream->offset += 8;
 
 	/* Reallocate the buffer if needed */
-	if (chunk->length > stream->buflen) {
-		stream->buffer = my_realloc(stream->buffer, chunk->length,
-					    MYF(MY_WME));
-		if (stream->buffer == NULL) {
+	if (chunk->length > chunk->buflen) {
+		chunk->data = my_realloc(chunk->data, chunk->length,
+					 MYF(MY_WME | MY_ALLOW_ZERO_PTR));
+		if (chunk->data == NULL) {
 			msg("xb_stream_read_chunk(): failed to increase buffer "
 			    "to %lu bytes.\n", (ulong) chunk->length);
 			goto err;
 		}
-		stream->buflen = chunk->length;
+		chunk->buflen = chunk->length;
 	}
 
 	/* Checksum */
 	F_READ(tmpbuf, 4);
```

Comment:
Previous blocks followed pattern:
```
read
process
update_offset
```
but not this time, it looks like `stream->offset` is not updated here.